src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -29,19 +32,134 @@ case class NoOpFingerprint() {
 }
 
 // IndexLogEntry-specific Content that uses IndexLogEntry-specific fingerprint.
-case class Content(root: String, directories: Seq[Content.Directory])
+case class Content(root: Directory) {
+
+  // List of fully qualified paths of all files mentioned in this Content object.
+  @JsonIgnore
+  lazy val files: Seq[Path] = {
+    // Recursively append parent dir path to subdirs and files.
+    def rec(prefixPath: Path, directory: Directory): Seq[Path] = {
+      val files = directory.files.map(f => new Path(prefixPath, f.name))
+      files ++ directory.subDirs.flatMap { dir =>
+        rec(new Path(prefixPath, dir.name), dir)
+      }
+    }
+
+    rec(new Path(root.name), root)
+  }
+}
+
 object Content {
-  case class Directory(path: String, files: Seq[FileInfo], fingerprint: NoOpFingerprint)
-  object Directory {
-    // modifiedTime is an Epoch time in milliseconds. (ms since 1970-01-01T00:00:00.000 UTC).
-    case class FileInfo(name: String, size: Long, modifiedTime: Long)
-    object FileInfo {
-      def apply(s: FileStatus): FileInfo =
-        FileInfo(s.getPath.toString, s.getLen, s.getModificationTime)
+  // Create a Content object from a directory path by recursively listing its leaf files. All
+  // files from the directory tree will be part of the Content.
+  def apply(path: Path): Content = Content(Directory(path))
```

Comment:
Can we rename this to `fromPath` to make the intention clear?

Reply:
thanks @imback82 , done